mqtt_client.c

```diff
@@ -958,6 +958,12 @@ esp_err_t esp_mqtt_client_reconnect(esp_mqtt_client_handle_t client)
 esp_err_t esp_mqtt_client_stop(esp_mqtt_client_handle_t client)
 {
     if (client->run) {
+        // Notify the broker we are disconnecting
```

Comment:
as @tuanpmt mentioned, this part should be protected by locks, because `esp_mqtt_client_stop` could be called from another task.

I can add those `MQTT_API_LOCK_FROM_OTHER_TASK` in a separate commit, if you're having issues with rebasing, but please let me know.

Please note that you've probably missed `push force` to your original branch while rebasing. Also your changes were not based on the top of `idf` branch

Reply:
@david-cermak I figured out why my rebase was failing:  I was rebasing my local branch and not the `origin` branch.  Rebasing the local branch and pushing results in the `origin` branch resulted in a merge.  

Anyway, I can add a lock.  Where can I find an example or documentation on how to use `MQTT_API_LOCK_FROM_OTHER_TASK`?

Reply:
I'm afraid there's no documentation, but you can look at any API which could be executed from user context, for example `esp_mqtt_client_publish`.
Just enclose the code with `MQTT_API_LOCK_FROM_OTHER_TASK(client);` and `MQTT_API_UNLOCK_FROM_OTHER_TASK(client);` macros.

Reply:
@david-cermak That seems easy enough.  I'll make the changes and then squash the two commits back into one.